dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/UCLouvainThesisReviewAction.java

```diff
@@ -56,9 +82,55 @@ public ActionResult execute(Context c, XmlWorkflowItem wfi, Step step, HttpServl
     }
 
     /**
-     * Process result when option {@link this#SUBMIT_REJECT} is selected.
-     * - Sets the reason and workflow step responsible on item in dc.description.provenance
-     * - Send workflow back to the submission
+     * Get the list of options available for this action.
+     * This is also used for rendering buttons on the frontend.
+     */
+    @Override
+    public List<String> getOptions() {
+        List<String> options = new ArrayList<String>();
+        options.add(SUBMIT_APPROVE);
+        options.add(SUBMIT_APPROVE_WITHOUT_DIFFUSION);
+        options.add(SUBMIT_REJECT);
+        options.add(RETURN_TO_POOL);
+        return options;
+    }
+
+    /**
+     * Process result when option 'SUBMIT_APPROVE_WITHOUT_DIFFUSION' is selected.
+     * - First delete all bitstream restriction and add only administrator access.
+     * - Add a new tag to keep a trace of the operation in the 'dc.description.X' metadata field.
+     * If reason is not given => error
+     */
+    public ActionResult processAcceptWithoutDiffusion(Context ctx, XmlWorkflowItem wfi, HttpServletRequest request) throws SQLException, AuthorizeException {
+        Item currentItem = wfi.getItem();
+        ItemService currentItemService = currentItem.getItemService();
+        String reason = request.getParameter(REJECT_REASON);
+
+        System.out.println("REASON: " + reason);
+
+        // 1. Change bitstreams access to admin only
+        List<Bitstream> itemBitstreams = this.getBitstreamsForItem(currentItem);
+        Group adminGroup = this.groupService.findByName(ctx, "Administrator");
+        if (!itemBitstreams.isEmpty() && adminGroup != null){
+            // Retrieve admin group 
+            for (Bitstream bitstream: this.getBitstreamsForItem(currentItem)){
+                this.restrictBitstream(ctx, bitstream, adminGroup);
+            }
+        }
+
+        // TODO: ADD TAG && USE REASON
+        // 2. Add provenance && a custom message tag for the item in "dc.description.diffusion"
+        this.addProvenance(ctx, currentItem, " Approved with no diffusion for entry into archive by ");
```

Comment:
Remove spaces from `message` param. Spaces must be added by method creating the full message.